src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsRequestBodyFilter.cs

```diff
@@ -13,97 +13,165 @@ public class XmlCommentsRequestBodyFilter : IRequestBodyFilter
         public XmlCommentsRequestBodyFilter(XPathDocument xmlDoc) : this(XmlCommentsDocumentHelper.CreateMemberDictionary(xmlDoc))
         {
         }
-
         internal XmlCommentsRequestBodyFilter(IReadOnlyDictionary<string, XPathNavigator> xmlDocMembers)
         {
             _xmlDocMembers = xmlDocMembers;
         }
 
         public void Apply(OpenApiRequestBody requestBody, RequestBodyFilterContext context)
         {
-            var parameterDescription =
-                context.BodyParameterDescription ??
-                context.FormParameterDescriptions.FirstOrDefault((p) => p is not null);
+            var bodyParameterDescription = context.BodyParameterDescription;
 
-            if (parameterDescription is null)
+            if (bodyParameterDescription is not null)
             {
-                return;
+                var propertyInfo = bodyParameterDescription.PropertyInfo();
+                if (propertyInfo is not null)
+                {
+                    ApplyPropertyTagsForBody(requestBody, context, propertyInfo);
+                    return;
+                }
+                var parameterInfo = bodyParameterDescription.ParameterInfo();
+                if (parameterInfo is not null)
+                {
+                    ApplyParamTagsForBody(requestBody, context, parameterInfo);
+                    return;
+                }
```

Comment:
I think it would be better to put the second part into an `else`, then we don't need the two `return`s.